.github/workflows/rock-update.yaml

```diff
@@ -93,7 +97,18 @@ jobs:
         run: |
           version="${{ steps.check.outputs.version }}"
           go_version=$(grep -Po "^go \K(\S+)" $GITHUB_WORKSPACE/application-src/go.mod) \
-          yq -i '.parts.${{ inputs.rock-name }}["build-snaps"] = ["go/" + strenv(go_version) + "/stable"]' $GITHUB_WORKSPACE/main/$version/rockcraft.yaml
+          # Delete the Go dependency and add the updated one
+          yq -i 'del(.parts.${{ inputs.rock-name }}.build-snaps.[] | select(. == "go/*"))' $GITHUB_WORKSPACE/main/$version/rockcraft.yaml
+          go_v="$go_version" yq -i '.parts.${{ inputs.rock-name }}.build-snaps += strenv(go_v)' $GITHUB_WORKSPACE/main/$version/rockcraft.yaml
+
+      - name: Update other build dependencies
+        if: ${{ steps.check.outputs.release != '' && inputs.update-script != '' }}
+        shell: bash
+        run: |
+          version="${{ steps.check.outputs.version }}"
+          application_src=$GITHUB_WORKSPACE/application-src
+          rockcraft_yaml=$GITHUB_WORKSPACE/main/$version/rockcraft.yaml
+          eval "${{ inputs.update-script }}"
```

Comment:
- Why `eval` and not `sh -c`?
- Mention in the description that:
  - this needs to be a one-liner?
  - the update command can use the version, app_src, rockcraft_yaml vars?

Reply:
`eval` instead of `sh -c` because this does **not** need to be a one-liner; the update **script** will:
1. fetch all the correct external dependencies versions from the appropriate files (e.g. for Prometheus, only Node from *web/ui/package.json*)
2. substitute those versions into the *rockcraft.yaml* file

The script needs access to the upstream project for 1 and the rockcraft file for 2; there are of course a bunch of ways to address this, but each has downsides: 
* pass them as arguments to the script? They won't be named and the per-repo script will be less readable;
* do not pass them at all, re-clone the repo in the update script, and somehow return the changes in a variable? Clunky